lessons/unittest/original/module.py

```diff
@@ -0,0 +1,35 @@
+''' 
+  A class we can use to test unittesting
+'''
+
+import numpy as np
+
+
+class Var(object):
+  ''' A container class for numpy arrays and some meta data '''
+
+  def __init__(self, name, data, axes=None, units=None):
+    ''' construct variable object with meta data '''
+    self.name = name
+    self.units = units
+    self.axes = axes
+    self.data = data
+    
+  def __iadd__(self, data):
+    ''' overload in-place addition function: add number or arrays '''
+    self.data += data
+    return self
+  
+  def __add__(self, other):
+    ''' overload addition function: add arrays and create new variable '''
+    # check input
+    if not isinstance(other, Var): raise Exception
```

Comment:
While the documentation seems to say that, I don't see the utility in that... does this not unnecessarily delay the error and make debugging harder? I would want to see an exception as early as possible, and ideally with a somewhat informative error message (e.g. printing str(Var) )